src/dfastbe/bank_erosion/data_models/inputs.py

```diff
@@ -268,6 +276,234 @@ def _read_river_axis(self) -> LineString:
+            "velocity": param_resolved["vship"],
+            "number": param_resolved["nship"],
+            "num_waves": param_resolved["nwave"],
+            "draught": param_resolved["draught"],
+            "type": param_resolved["shiptype"],
+            "slope": param_resolved["slope"],
+            "reed": param_resolved["reed"],
+        }
+
+        return cls(config_file, **param_dict)
+
+    @staticmethod
+    def _get_param(
+        config_file: ConfigFile, name: str, num_stations_per_bank, iq_str="", **kwargs
+    ):
+        """Get a parameter from the configuration file.
+
+        Args:
+            config_file (ConfigFile):
+                Configuration file containing parameters.
+            name (str):
+                Name of the parameter to retrieve.
+            num_stations_per_bank (List[int]):
+                The number of stations per bank.
+            iq_str (str, optional):
+                String to append to the parameter name for specific levels.
+            **kwargs:
+                Additional keyword arguments for parameter retrieval.
+        Returns:
+            np.ndarray: The parameter value as a NumPy array.
+        """
+        return config_file.get_parameter(
+            "Erosion",
+            f"{name}{iq_str}",
+            num_stations_per_bank,
+            **kwargs,
+        )
+
+    @classmethod
+    def _get_initial_parameter_definitions(cls) -> List[Parameters]:
+        """Get parameter definitions for discharge parameters.
+
+        Returns:
+            List[namedtuple]: List of parameter definitions.
+        """
+        return [
+            cls.Parameters("VShip", None, None, True, True, None),
+            cls.Parameters("NShip", None, None, True, True, None),
+            cls.Parameters("NWave", 5, None, True, True, None),
+            cls.Parameters("Draught", None, None, True, True, None),
+            cls.Parameters("ShipType", None, [1, 2, 3], True, None, None),
+            cls.Parameters("Slope", 20, None, None, True, "slp"),
+            cls.Parameters("Reed", 0, None, None, True, "rdd"),
+        ]
+
+    def _get_discharge_parameter_definitions(self) -> List[Parameters]:
+        """Get parameter definitions for discharge parameters.
+
+        Returns:
+            List[namedtuple]: List of parameter definitions.
+        """
+        return [
+            self.Parameters("VShip", self.velocity, None, None, None, None),
+            self.Parameters("NShip", self.number, None, None, None, None),
+            self.Parameters("NWave", self.num_waves, None, None, None, None),
+            self.Parameters("Draught", self.draught, None, None, None, None),
+            self.Parameters("ShipType", self.type, [1, 2, 3], True, None, None),
+            self.Parameters("Slope", self.slope, None, None, True, "slp"),
+            self.Parameters("Reed", self.reed, None, None, True, "rdd"),
+        ]
+
+    @staticmethod
+    def _calculate_ship_derived_parameters(
```

Comment:
can you revise the protected methods, here and if they are called from the Erosion class convert them to normal methods, only protected if they are called inside the `ShipsParameters` itself.